cli/src/commands/migrateToInQueries.js

```diff
@@ -0,0 +1,269 @@
+    visualisation.axesyQuery = newAxesyQuery;
+    logger.info('convert axesyQuery');
+    logger.info(oldAxesyQuery);
+    logger.info(newAxesyQuery);
+
+    visualisation.hasBeenMigrated = true;
+    await visualisation.save();
+  }
+};
+
+const convertDashboards = async () => {
+  const dashboards = await Dashboard.find({});
+
+  for (const dashboard of dashboards) {
+    if (dashboard.hasBeenMigrated) {
+      logger.warn(`dashboard (${dashboard._id}) has been migrated`);
+      return;
+    }
+
+    const oldShareableList = lodash.cloneDeep(dashboard.shareable);
+
+    // mutate shareable
+    dashboard.shareable.forEach((shareable) => {
+      const newFilter = buildNewQuery(shareable.filter);
+      shareable.oldFilter = shareable.filter;
+      shareable.filter = newFilter;
+    });
+
+    const newShareableList = lodash.cloneDeep(dashboard.shareable);
+
+    // Save updated queries and log
+    logger.info(`Update Dashboard (${dashboard._id})`);
+    lodash.zip(oldShareableList, newShareableList)
+      .forEach(([oldShareable, newShareable]) => {
+        if (oldShareable.filter !== newShareable.filter) {
+          logger.info('convert shareable filter');
+          logger.info(oldShareable.filter);
+          logger.info(newShareable.filter);
+        }
+      });
+
+    dashboard.hasBeenMigrated = true;
+    await dashboard.save();
+  }
+};
+
+const convertStatementForwarding = async () => {
+  const statementForwardings = await StatementForwarding.find();
+
+  for (const statementForwarding of statementForwardings) {
+    if (statementForwarding.hasBeenMigrated) {
+      logger.warn(`statementForwarding (${statementForwarding._id}) has been migrated`);
+      return;
+    }
+
+    const oldQuery = statementForwarding.query;
+    const newQuery = buildNewQuery(oldQuery);
+
+    statementForwarding.oldQuery = statementForwarding.query;
+    statementForwarding.query = newQuery;
+
+    logger.info(`Update StatementForwarding (${statementForwarding._id})`);
+    logger.info('convert query');
+    logger.info(oldQuery);
+    logger.info(newQuery);
+
+    statementForwarding.hasBeenMigrated = true;
+    await statementForwarding.save();
+  }
+};
+
+const convertUsers = async () => {
+  const users = await User.find();
+
+  for (const user of users) {
+    if (user.hasBeenMigrated) {
+      logger.warn(`user (${user._id}) has been migrated`);
+      return;
+    }
+
+    logger.info(`Update User (${user._id})`);
+
+    user.organisationSettings.forEach((organisationSetting) => {
+      const oldFilter = organisationSetting.filter;
+      const newFilter = buildNewQuery(oldFilter);
+
+      organisationSetting.oldFilter = organisationSetting.filter;
+      organisationSetting.filter = newFilter;
+
+      logger.info('convert organisationSettings filter');
+      logger.info(oldFilter);
+      logger.info(newFilter);
+    });
+
+    user.hasBeenMigrated = true;
+    await user.save();
```

Comment:
I suspect @ht2 and @asahd might prefer us to use bulk ops, but given the small size of these collections this looping and awaiting might be ok. What do you think @ht2 and @asahd?

Reply:
Should be fine, only a few hundred of these to worry about and does mean each one is done atomically

Reply:
Okay nice thanks, yeah agreed 👍